spec/models/miq_expression/miq_expression_spec.rb

```diff
@@ -1,12 +1,136 @@
+    it "generates the SQL for an EQUAL expression" do
+      sql, * = MiqExpression.new("EQUAL" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name = 'foo'")
+    end
+
+    it "generates the SQL for a = expression" do
+      sql, * = MiqExpression.new("=" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name = 'foo'")
+    end
+
+    it "generates the SQL for a LIKE expression" do
+      sql, * = MiqExpression.new("LIKE" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name LIKE '%foo%'")
+    end
+
+    it "generates the SQL for a NOT LIKE expression" do
+      sql, * = MiqExpression.new("NOT LIKE" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("!(vms.name LIKE '%foo%')")
+    end
+
+    it "generates the SQL for a STARTS WITH expression " do
+      sql, * = MiqExpression.new("STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name LIKE 'foo%'")
+    end
+
+    it "generates the SQL for an ENDS WITH expression" do
+      sql, * = MiqExpression.new("ENDS WITH" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name LIKE '%foo'")
+    end
+
+    it "generates the SQL for an INCLUDES" do
+      sql, * = MiqExpression.new("INCLUDES" => {"field" => "Vm-name", "value" => "foo"}).to_sql
+      expect(sql).to eq("vms.name LIKE '%foo%'")
+    end
+
+    it "generates the SQL for an AND expression" do
+      exp1 = {"STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}}
+      exp2 = {"ENDS WITH" => {"field" => "Vm-name", "value" => "bar"}}
+      sql, * = MiqExpression.new("AND" => [exp1, exp2]).to_sql
+      expect(sql).to eq("(vms.name LIKE 'foo%' AND vms.name LIKE '%bar')")
+    end
+
+    it "generates the SQL for an AND expression where only one is supported by SQL" do
+      exp1 = {"STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}}
+      exp2 = {"ENDS WITH" => {"field" => "Vm-platform", "value" => "bar"}}
+      sql, * = MiqExpression.new("AND" => [exp1, exp2]).to_sql
+      expect(sql).to eq("(vms.name LIKE 'foo%')")
+    end
+
+    it "returns nil for an AND expression where none is supported by SQL" do
+      exp1 = {"STARTS WITH" => {"field" => "Vm-platform", "value" => "foo"}}
+      exp2 = {"ENDS WITH" => {"field" => "Vm-platform", "value" => "bar"}}
+      sql, * = MiqExpression.new("AND" => [exp1, exp2]).to_sql
+      expect(sql).to be_nil
+    end
+
+    it "generates the SQL for an OR expression" do
+      exp1 = {"STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}}
+      exp2 = {"ENDS WITH" => {"field" => "Vm-name", "value" => "bar"}}
+      sql, * = MiqExpression.new("OR" => [exp1, exp2]).to_sql
+      expect(sql).to eq("(vms.name LIKE 'foo%' OR vms.name LIKE '%bar')")
+    end
+
+    it "returns nil the SQL for an OR expression where one is not supported by SQL" do
```

Comment:
@gtanzillo is this correct? I expected this to behave like `AND`, where it returns the sub-expression that is supported by SQL